src/openvswitch.c

```diff
@@ -29,6 +30,48 @@
 #include "util.h"
 #include "util-internal.h"
 
+#ifndef PREFIX
+#define PREFIX "/usr/"
+#endif
+
+#ifndef BINDIR
+#define BINDIR PREFIX"bin"
+#endif
+
+#ifndef SNAPBINDIR
+#define SNAPBINDIR "/snap/bin"
+#endif
+
+#define OPENVSWITCH_OVS_VSCTL "ovs-vsctl"
+
+static char netplan_openvswitch_ovs_vsctl_path[
+    MAX(sizeof BINDIR + 1 + sizeof OPENVSWITCH_OVS_VSCTL,
+        sizeof SNAPBINDIR + 1 + sizeof OPENVSWITCH_OVS_VSCTL)];
+
+void
+netplan_openvswitch_vsctl__(void)
+{
+    if (g_file_test(SNAPBINDIR"/"OPENVSWITCH_OVS_VSCTL, G_FILE_TEST_EXISTS)) {
+        snprintf(netplan_openvswitch_ovs_vsctl_path,
+                 sizeof netplan_openvswitch_ovs_vsctl_path,
+                 SNAPBINDIR"/"OPENVSWITCH_OVS_VSCTL);
+    } else {
+        snprintf(netplan_openvswitch_ovs_vsctl_path,
+                 sizeof netplan_openvswitch_ovs_vsctl_path,
+                 BINDIR"/"OPENVSWITCH_OVS_VSCTL);
+    }
+
+}
+
+const char *
+netplan_openvswitch_ovs_vsctl(void)
+{
+    if (!*netplan_openvswitch_ovs_vsctl_path) {
```

Comment:
Can we be sure this is always null-initialized? Some architectures/compilers might not do that for us. We could probably do that above, or use a higher level concept, like glib's [GString](https://docs.gtk.org/glib/struct.String.html), that takes care of that for us.

Another thought/nitpick: What happens to a longer-running Netplan process (e.g. a daemon linking to `libnetplan`), where the OVS snap gets installed midway through the Netplan process' lifetime? I think it would continue calling into the system-OVS, which might or might not be OK. Granted, that is more of a future topic, as the OVS generator is mostly executed in isolation currently.